### DISCUSSION TITLE
Config.Json missing causing a loop?

### DISCUSSION THREAD

**Discussion by Iamthemedic** [tsightler/ring-mqtt#866] [Support Q&A] [answered]
Hello!

I'm using ring-mqtt via docker compose; noticed when doing some recent poking that it is looping a restarting status. Checked the logs via portainer and I'm getting this

`2024-06-30T15:02:37.305Z ring-mqtt No configuration file found at /data/config.json`
`2024-06-30T15:02:37.305Z ring-mqtt Please map a persistent volume to this location and place a configuration file there.`

I used docker compose to do this. The portion of my compose yaml for ring-mqtt is below (exact copy paste from the setup page):

>  ring-mqtt:
>     container_name: ring-mqtt
>     restart: unless-stopped
>     image: tsightler/ring-mqtt
>     ports:
>       - 8554:8554                      # Enable RTSP port for external media player access
>     volumes:
>       - /opt/ring-mqtt:/data           # Mapping of local folder to provide persistent storage
>     logging:                           #limit logs to 10m and 3 files
>       options:
>         max-size: 10m
>         max-file: "3"

When looking at the folder in terminal, there is nothing in the /opt/ring-mqtt folder at all. I completely removed the container and did a fresh docker compose up -d and issue is same.  I also cannot find anything about setup of the config.json in the help articles so a little lost on what's going on here, hoepfully someone can help! :)

**Comment by tsightler** [comment, marked as answer]
Documentation for docker installation covers the steps required to perform authentication and create the initial configuration file, so I'm not really sure why you are unable to find anything about it, it is part of the required authentication step.  If you started fresh, then just follow those steps as documented.

**Reply by Iamthemedic** [reply]
I was able to get this sorted via reading top to bottom. I assumed (my bad) that the docker compose was somehow able to bypass some of this and would prompt downstream. Maybe worthwhile adding a disclaimer (I'm a somewhere between wet behind the ears and not quite a total noob on my Docker knowledge, and very new on MQTT).  There's an overwhelming amount of information out there so just trying to keep it all straight. Thanks for the prompt reply, and tolerating my noobiness!
